### PR TITLE
Rewrote member DO availability

### PR DESCRIPTION
**Pull request by ctaylor451**

### CODE REVIEW

**Comment by ctaylor451** [comment]
@kduncklee not sure why the build is failing is, seems to want a phone number for sms send. Haven't seen this before

Please check the python for style and correctness

File: main/views/api_views.py [hunk 1]
```diff
@@ -128,6 +130,44 @@ class DoViewSet(BaseViewSet):
...
+            return Response(content, status=status.HTTP_404_NOT_FOUND)
+
+        # If the request is a member and a specific quarter,
+        # create all the objects for that quarter
+        #import pdb; pdb.set_trace()
+        if member is not None and year is not None and quarter is not None:
+            # Use the first object to compute number of weeks in the quarter
+            avail1, created = DoAvailable.objects.get_or_create(
+                member=member, year=year, quarter=quarter, week=1)
+            for week in avail1.weeks(year, quarter):
```
Comment (kduncklee) [inline main/views/api_views.py]: weeks is a class method. You can just do DoAvailable.weeks(...) and not special case the first one.

File: main/views/api_views.py [hunk 2]
```diff
@@ -128,6 +130,44 @@ class DoViewSet(BaseViewSet):
                      'comment', 'member', )
     search_fields = ('member__username',)
 
+    def list(self, request, *args, **kwargs):
+        id = request.query_params.get('member', None)
+        try:
+            member = Member.objects.filter(id=id)[0]
+        except:
```
Comment (kduncklee) [inline main/views/api_views.py]: Generally better to list the exception you're catching, but this is fine for now.

Comment (ctaylor451) [reply main/views/api_views.py]: In this case any error probably indicates a bad parameter

**Pull request merged**